lib/plugins/aws/package/compile/events/s3/index.js

```diff
@@ -143,13 +166,14 @@ class AwsCompileS3Events {
       _.forEach(dependsOnToCreate, item => {
         const lambdaPermissionLogicalId = this.provider.naming.getLambdaS3PermissionLogicalId(
           item.functionName,
-          item.bucketName
+          bucketsLogicalIds[item.bucketName] || item.bucketName
```

Comment:
I think we don't need to do any change here

Reply:
The reason why I kept it this way is that I don't want the generated logical id to contain the bucket name. Assume you want to override the generated permission and have a custom bucket name containing the current `stage` ie: `my-custom-bucket-dev`. To override the generated Permission you have to specify this resource:
```
HelloLambdaPermissionMycustombucketdevS3:
      Type: 'AWS::Lambda::Permission'
      Properties:
```

But then it becomes problematic if you deploy a different stage as you'll have a different generated logical id

My point is that if you do custom s3 bucket configuration you want to avoid having dynamic bucket name in the logical id :)

Reply:
Ok, so as user, when defining bucketName on his own, can make it dynamic (e.g. adapt to current stage, region etc.). It's more convinient to align to resource id, which shouldn't change across those.

Sounds reasonable, thanks for explanation